src/main/java/edu/ucsb/cs156/happiercows/jobs/UpdateCowHealthJob.java

```diff
@@ -42,8 +42,14 @@ public void accept(JobContext ctx) throws Exception {
             for (UserCommons userCommons : allUserCommons) {
                 User user = userRepository.findById(userCommons.getUserId()).orElseThrow(() -> new RuntimeException("Error calling userRepository.findById(" + userCommons.getUserId() + ")"));
                 ctx.log("User: " + user.getFullName() + ", numCows: " + userCommons.getNumOfCows() + ", cowHealth: " + userCommons.getCowHealth());
-
                 var newCowHealth = calculateNewCowHealthUsingStrategy(cowHealthUpdateStrategy, commons, userCommons, totalCows);
+                
+                int temp = userCommons.getCowDeaths();
+                calculateCowDeaths(userCommons, ctx);
+                if (userCommons.getCowDeaths() > temp) {
+                    newCowHealth = 100.0;
+                }
+                
                 ctx.log(" old cow health: " + userCommons.getCowHealth() + ", new cow health: " + newCowHealth);
                 userCommons.setCowHealth(newCowHealth);
```

Comment:
It's still not working... the cow health is set on line 54, after `calculateCowDeaths` is called.

Reply:
It is, but it's being set as 100 on line 50. If calculateCowDeaths function runs, then newCowHealth = 100.0, which overrides the newCowHealth value from calculateNewCowHealthUsingStrategy.

Reply:
I meant calculateCowDeaths uses the health from userCommons, but that value isn't updated until line 54. calculateNewCowHealthUsingStrategy only returns the new health, it doesn't set it. calculateCowDeaths should use newCowHealth instead.

Reply:
Should be fine now